src/store/sampleExtraction/actions.js

```diff
@@ -1,13 +1,15 @@
-import handlePromise from '@/api/PromiseHelper'
+import { handleResponse } from '@/api/ResponseHelper'
+import deserialize from '@/api/JsonApi'
 
 const getSampleExtractionTubesForBarcodes = async ({ commit, getters }, barcodes) => {
   let barcodeString = barcodes.join(',')
   let request = getters.sampleExtractionTubeRequest
   let promise = request.get({ filter: { barcode: barcodeString } })
-  let response = await handlePromise(promise)
 
-  if (response.successful && !response.empty) {
-    let assets = response.deserialize.assets
+  const response = await handleResponse(promise)
+
+  if (response.success && response.data.data.length != 0) {
```

Comment:
It could be better to do the error handling in here rather than in the view.

...
if (response.success) {
  if (response.data.data.length === 0) {
    return { ...response, success: false, errors: 'Sample Extraction tubes failed to be imported' }
  }
 const assets = deserialize(response.data).assets
 commit('setSampleExtractionTubes', assets)
}
...